lib/dao.js

```diff
@@ -1597,6 +1598,22 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
@ssh24 
Sorry for the typo.

The actual code 2.x branch [/lib/dao.js#L1615](/strongloop/loopback-datasource-juggler//blob/2.x/lib/dao.js#L1615) thrown the that Error.

Adding the checking that @bajtos suggested should fix the issue. #1434 

Currently the actual code doesn't check if `...type.definition` exist.